### DISCUSSION TITLE
Populating table with JSON

### DISCUSSION THREAD

**Discussion by CZHach** [TanStack/table#2075] [General]
Hi, this component was easy to get up and running using the newPerson and makeData examples, but now I'm wondering, how would you populate this table with JSON from an API call?  I can't seem to find any examples.  

Thanks!

**Comment by fnn** [comment]
Hey,

the same way. React-table does not handle API calls for you. Instead of calling `makeData()` you are doing a async fetch or call a selector to get your data. Depends on your setup. How are you loading data in your app?

Take a look at the controlled pagination example. There you can see how to handle pagination if the page count is comming from the API request (but still no real API request. It is just to demonstrate).

**Reply by CZHach** [reply]
Thanks, right now, the table app (App.js) has these two lines which I assume are the two lines responsible for getting the data for the table:
```
    const [data, setData] = React.useState(() => makeData(1000))
    const [originalData] = React.useState(data)
```

So, in order to get JSON data in there from an API, would I replace the "makeData(1000)" part with something that would call an API to get JSON?

Thanks again
